hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileManager.java

```diff
@@ -50,15 +50,15 @@ public interface StoreFileManager {
    */
   @RestrictedApi(explanation = "Should only be called in StoreEngine", link = "",
       allowedOnPath = ".*(/org/apache/hadoop/hbase/regionserver/StoreEngine.java|/src/test/.*)")
-  void loadFiles(List<HStoreFile> storeFiles);
+  void loadFiles(List<HStoreFile> storeFiles) throws IOException;
```

Comment:
Why we need to add the exception declaration here?

Reply:
DefaultStoreFileManager needs to read the store files while loading them to find out if they are live or historical files. Doing this at the load time once eliminates it doing at the scan time every time. This was suggested during previous reviews. Reading the files requires handling IOException.

Reply:
Does it actually throw any exceptions? Especially for insertNewFiles and addCompactionResults, I think we should hava already loaded the related metadata in memory already?

Reply:
Ping.

Reply:
And also this one, could you please verify whether we actually throw any exceptions here? We'd better not declare throws if there is no actual exception thrown.

Reply:
Sorry I missed this one before. When DefaultStoreFileManager loads the store files, it also reads the file metadata to see if they are live or not in order to form the list of live store files. Reading file can generate IOException. Please see
```
private List<HStoreFile> getLiveFiles(Collection<HStoreFile> storeFiles) throws IOException {
    List<HStoreFile> liveFiles = new ArrayList<>(storeFiles.size());
    for (HStoreFile file : storeFiles) {
      file.initReader();
      if (!file.isHistorical()) {
        liveFiles.add(file);
      }
    }
    return liveFiles;
  }

  @Override
  public void loadFiles(List<HStoreFile> storeFiles) throws IOException {
    this.storeFiles = new StoreFileList(ImmutableList.sortedCopyOf(storeFileComparator, storeFiles),
      enableLiveFileTracking
        ? ImmutableList.sortedCopyOf(storeFileComparator, getLiveFiles(storeFiles))
        : null);
  }
```

Reply:
I know the there are throws declarations, what I mean here is whether it will actually throw any exceptions out. For example, in loadFiles, maybe we have already called initReader for all files so there is no problem? Just want to confirm this.

Reply:
I do not know the answer. I was going to debug and find the answer. However, I could not find a test that calls this method with non-empty storeFiles. I need to device a test where I need to generate some files and then close the store, and reopen it while debugging it. I will try to do it later.

Reply:
Anyway, not a blocker. Can debug later.